mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -3692,6 +3713,29 @@ static OpTy castOrGetParentOfType(Operation *op, bool immediateParent = false) {
   return op->getParentOfType<OpTy>();
 }
 
+static uint64_t getTypeByteSize(mlir::Type type, DataLayout dl) {
+      uint64_t sizeInBits = dl.getTypeSizeInBits(type);
+      uint64_t sizeInBytes = sizeInBits / 8;
+      return sizeInBytes;
+}
+
+template <typename OpTy>
+static uint64_t getReductionDataSize(OpTy &op) {
```

Comment:
Don't we currently need this for other reductions? If not, do we expect to use it elsewhere or is this actually only needed to be calculated for teams reductions within a target region?